.github/workflows/yki.yml

```diff
@@ -0,0 +1,70 @@
+name: YKI
+
+on:
+  workflow_dispatch:
+  push:
+    paths-ignore:
+      - '**.md'
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Prepare Clojure env
+        uses: ./.github/actions/prepare-clojure-env
+
+      - name: Run tests
+        run: lein test
+
+  build:
+    runs-on: ubuntu-latest
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Prepare Clojure env
+        uses: ./.github/actions/prepare-clojure-env
+
+      - name: Build uberjar
+        run: lein uberjar
+
+      - uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: target
+          key: ${{ github.sha }}
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: [test, build]
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: target
+          key: ${{ github.sha }}
+
+      - name: Build and deploy Docker image
+        shell: bash
+        run: |
+          git clone https://github.com/Opetushallitus/ci-tools.git
+          source ci-tools/common/setup-tools.sh
+          mv target/yki-*-standalone.jar $DOCKER_BUILD_DIR/artifact/yki.jar
+          cp -vr oph-configuration $DOCKER_BUILD_DIR/config/
+          export BASE_IMAGE="baseimage-fatjar-openjdk17:master"
+          ./ci-tools/common/pull-image.sh
+          ./ci-tools/github-build/build-fatjar.sh yki
+          ./ci-tools/common/clean-docker-build-dir.sh
```

Comment:
Tämä ei ilmeisesti poista kuitenkaan imagea, joka seuraavassa vaiheessa ECR:ään viedään? Tätä ei muuten kieli-ja-kaantajatutkinnot puolen common-deploy:ssa, voisi lisätä sinne jos hyödyllistä tehdä.

Reply:
Ei vissiin. Yritin hieman apinoida muista OPHn Clojure-projekteista tätä pohjaa ja toi loppuosa on aika sama kuin esim. https://github.com/Opetushallitus/maksut/blob/master/.github/workflows/build.yml

Reply:
Tuossa `maksut` osalta tosin ajetaan juuri tämä `./ci-tools/common/clean-docker-build-dir.sh` upload-image.sh:n jälkeen. Tosin jos toisaalla nämä laitettu toisin päin niin silloin nyt ei ainakaan väliä.

Reply:
Ai joo, näinpä tosiaan. Vaihdanpa nyt varmuuden vuoksi tässäkin samoin päin.